dali/operators/reader/loader/video_loader.cc

```diff
@@ -585,10 +590,16 @@ void VideoLoader::read_file() {
           if (!is_first_frame) {
             nonkey_frame_count = 0;
             if (frame > req.frame + req.count) {
-              // Found a key frame past the requested range. We can stop searching
+              // Found a second key frame past the requested range. We can stop searching
               // (If there were missing frames in the range they won't be found after
               // the next key frame)
-              break;
+              // in case HEVC it seems that preceding frames can appear after a given key frame
+              // but rather not after the next one
+              if (key_frames_count > key_frames_treshold) {
+                key_frames_count = 0;
+                break;
+              }
+              ++key_frames_count;
             }
```

Comment:
If my comment is right, this can be replaced with (note the 2):
```
const int key_frames_treshold = 2;

if (frame > req.frame + key_frames_treshold * req.count) {
   break;
}
```
Which is probably simpler.

Reply:
`key_frames_treshold * req.count` doesn't make any sense. `req.count` has nothing to do with keyframes.